future_test.go

```diff
@@ -236,23 +241,61 @@ func TestFutureGetIteratorError(t *testing.T) {
 
 func TestFutureSetStateRaceCondition(t *testing.T) {
 	err := errors.New("any error")
-	resp := &Response{}
 
 	for i := 0; i < 1000; i++ {
-		fut := NewFuture()
+		fut := NewFuture(test_helpers.NewMockRequest())
 		for j := 0; j < 9; j++ {
 			go func(opt int) {
 				if opt%3 == 0 {
-					respAppend := &Response{}
-					fut.AppendPush(respAppend)
+					fut.AppendPush(Header{}, nil)
 				} else if opt%3 == 1 {
 					fut.SetError(err)
 				} else {
-					fut.SetResponse(resp)
+					fut.SetResponse(Header{}, nil)
 				}
 			}(j)
 		}
 	}
 	// It may be false-positive, but very rarely - it's ok for such very
 	// simple race conditions tests.
 }
+
+func TestFutureGetIteratorIsPush(t *testing.T) {
+	fut := NewFuture(test_helpers.NewMockRequest())
+	fut.AppendPush(Header{}, nil)
+	fut.SetResponse(Header{}, nil)
+	it := fut.GetIterator()
+
+	assert.True(t, it.IsPush())
+	it.Next()
+	assert.False(t, it.IsPush())
+}
+
+func TestFuture_Get(t *testing.T) {
+	resp := test_helpers.NewMockResponse(t, []interface{}{'v', '2'})
+
+	fut := NewFuture(test_helpers.NewMockRequest())
+	fut.SetResponse(Header{}, bytes.NewBuffer(resp.GetData()))
+
+	data, err := fut.Get()
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(data))
+	assert.Equal(t, 'v', data[0])
+	assert.Equal(t, '2', data[1])
+}
+
+func TestFuture_GetResponse(t *testing.T) {
+	mockResp := test_helpers.NewMockResponse(t, []interface{}{'v', '2'})
+
+	fut := NewFuture(test_helpers.NewMockRequest())
+	fut.SetResponse(Header{}, bytes.NewBuffer(mockResp.GetData()))
+
+	resp, err := fut.GetResponse()
+	assert.NoError(t, err)
+
```

Comment:
Yes, but it will work like this:
```go
resp, err := fut.GetResponse()
assert.NoError(t, err)
respConv, ok := resp.(*test_helpers.MockResponse)
assert.True(t, ok)
assert.Equal(t, mockResp, *respConv)
```